src/api/earmark.ts

```diff
@@ -22,3 +23,17 @@ export const earmarkGasPrice = (gasPrice: string, userPrint: string): string =>
 
 // simple concatenation, with '0x' for empty data to have `0x<userPrint>` at the least
 export const earmarkTxData = (data = '0x', userPrint: string): string => data + userPrint
+
+const GAS_PER_DATA_CHAR = 8 // wei
```

Comment:
Interesting.

I know about the issue with the cancellation lacking gas, but was this much extra data also causing problems?

Reply:
We estimated before the earmark. Cause the gas estimation override happens before the transaction is sent.
The earmark has some costs. It's the responsibility of the middleware that adds some extra data, to adjust the gas to cover the cost of the new data

Reply:
The gas difference in our tests was:
`72185 - 72073 = 112`

`112 / 14 digits = 8 gas per digit`

This is how we arrived at the magic number

Reply:
Not questioning the number, just asking whether 112 gas might cause the tx to fail

Reply:
if I understand the question, yes. This missing 112 would make the tx fail. Cause it really needs this gas, so the tx cannot complete